docs/site/migration/auth/passport.md

```diff
@@ -10,8 +10,379 @@ permalink: /doc/en/lb4/migration-auth-passport.html
+  credentials in the request. Include the `verify` function in the provider
+  class.
+
+```ts
+  /**
+   * authenticate user with provided username and password
+   *
+   * @param username
+   * @param password
+   * @param done
+   *
+   * @returns User model
+   */
+  verify(
+    username: string,
+    password: string,
+    done: (error: any, user?: any) => void,
+  ): void {
+    this.userRepository
+      .find({
+        where: {
+          email: username,
+        }
+      })
+      .then((users: User[]) => {
+        const user = users[0];
+        if (!user.credentials || user.credentials.password !== password) {
+          return done(null, false);
+        }
+        // Authentication passed, return user profile
+        done(null, user);
+      })
+      .catch(err => {
+        done(err);
+      });
+  }
+```
+
+- Bind the strategy provider class to the application
+
+```ts
+export class UserApplication extends BootMixin(
+  ServiceMixin(RepositoryMixin(RestApplication)),
+) {
+  constructor(options: ApplicationConfig = {}) {
+    super(options);
+
+    this.add(createBindingFromClass(BasicStrategy));
+
+    this.sequence(MySequence);
+    this.component(AuthenticationComponent);
+  }
+}
+```
+
+## OAuth2 Strategies
+
+- how to use passport strategies for oauth2 authorization flow with external
+  social apps like Facebook.
+- this involves redirecting to an external app and user entering credentials in
+  that app's login page.
+- this usecase includes multiple phases of authentication.
+
+### Configuring Authentication Endpoints
+
+- For OAuth2 authorizaton flow, we need authentication endpoints that
+  participate to get the user validated with an external system.
+- This essentially means the `@authenticate` decorator is used with different
+  sematics compared to the non-oauth2 section above.
+- Here the controller methods become small parts of the larger oauth2 dialog.
+- We ideally create a controller with two endpoints decorated with
+  `@authenticate(`{passport-strategy-name}`)`
+- One of the endpoints is for redirecting to the external provider app and the
+  other is for getting called back by the external app.
+
+  - Create a controller with authentication endpoints as in below example:
+
+    - A method to redirect to the third party app (method `loginToThirdParty` in
+      the below example)
+
+      - an endpoint for api clients to login via a third party app
+      - the passport strategy identifies this call as a redirection to third
+        party
+      - this endpoint redirects to the third party authorization url
+
+    - A method for the third Party app to callback
+      - this is the callback for the thirdparty app (method `thirdPartyCallBack`
+        in the below example)
+      - on successful user login the third party calls this endpoint with an
+        access code
+      - the passport oauth2 strategy exchanges the code for an access token
```

Comment:
`oauth2` -> `OAuth2` - similar for other instances. 

```suggestion
      - the passport OAuth2 strategy exchanges the code for an access token
```

Reply:
agreed